cli/azd/pkg/async/task.go

```diff
@@ -0,0 +1,149 @@
+package async
+
+// Task represents a long running async operation
+type Task[R comparable] struct {
+	isComplete    bool
+	hasResult     bool
+	result        R
+	resultChannel chan R
+	error         error
+}
+
+// Runs the specified taskFn as a go routine
+func (t *Task[R]) Run(taskFn TaskRunFunc[R]) {
```

Comment:
An important thing of `async` is to support Cancellation.  Is there a way to handle it from the Task?  or does it depend on the taskFn directly?

The Run() function should take a `context.Context` that we can use to cancel the operation, either on timeout or on demand.
There should probably be a `Cancel()` function for a Task to perform on-demand cancellation.

A similar `task` concept in the AzureSDK libraries is el `LRO` (long running operation).  The Go lib uses a Poller to implement it (https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/runtime/poller.go#L185)

For the go skd lib, the operation polls status from a server instead of waiting for a channel to be closed.

Reply:
Good feedback, thanks!  I will add in better support for cancellation.

Reply:
Added #442 to track this.